From 330e7257464b405d0f0edfd6d87eddc51c96c67f Mon Sep 17 00:00:00 2001
From: Ondrej Kubik <ondrej.kubik@canonical.com>
Date: Wed, 21 Jun 2017 17:23:58 +0000
Subject: [PATCH 1/1] suport patch for dragonboard410c for ubuntu-core patch
 needs to revert changes to driver/mmc/sdhci.c which breaks support for emmc

Signed-off-by: Ondrej Kubik <ondrej.kubik@canonical.com>
---
 configs/dragonboard410c_defconfig |  2 ++
 drivers/mmc/sdhci.c               | 15 ++++-----------
 include/configs/dragonboard410c.h | 35 ++++++++++++++++++++++++-----------
 3 files changed, 30 insertions(+), 22 deletions(-)

diff --git a/configs/dragonboard410c_defconfig b/configs/dragonboard410c_defconfig
index 90c145a..02f6bdb 100644
--- a/configs/dragonboard410c_defconfig
+++ b/configs/dragonboard410c_defconfig
@@ -37,3 +37,5 @@ CONFIG_USB_EHCI_MSM=y
 CONFIG_USB_ULPI_VIEWPORT=y
 CONFIG_USB_ULPI=y
 CONFIG_USB_STORAGE=y
+CONFIG_CMD_MD5SUM=y
+CONFIG_SYS_NO_FLASH=y
diff --git a/drivers/mmc/sdhci.c b/drivers/mmc/sdhci.c
index b745977..c94d58d 100644
--- a/drivers/mmc/sdhci.c
+++ b/drivers/mmc/sdhci.c
@@ -72,7 +72,6 @@ static int sdhci_transfer_data(struct sdhci_host *host, struct mmc_data *data,
 				unsigned int start_addr)
 {
 	unsigned int stat, rdy, mask, timeout, block = 0;
-	bool transfer_done = false;
 #ifdef CONFIG_MMC_SDHCI_SDMA
 	unsigned char ctrl;
 	ctrl = sdhci_readb(host, SDHCI_HOST_CONTROL);
@@ -90,23 +89,17 @@ static int sdhci_transfer_data(struct sdhci_host *host, struct mmc_data *data,
 			       __func__, stat);
 			return -EIO;
 		}
-		if (!transfer_done && (stat & rdy)) {
+		if (stat & rdy) {
 			if (!(sdhci_readl(host, SDHCI_PRESENT_STATE) & mask))
 				continue;
 			sdhci_writel(host, rdy, SDHCI_INT_STATUS);
 			sdhci_transfer_pio(host, data);
 			data->dest += data->blocksize;
-			if (++block >= data->blocks) {
-				/* Keep looping until the SDHCI_INT_DATA_END is
-				 * cleared, even if we finished sending all the
-				 * blocks.
-				 */
-				transfer_done = true;
-				continue;
-			}
+			if (++block >= data->blocks)
+				break;
 		}
 #ifdef CONFIG_MMC_SDHCI_SDMA
-		if (!transfer_done && (stat & SDHCI_INT_DMA_END)) {
+		if (stat & SDHCI_INT_DMA_END) {
 			sdhci_writel(host, SDHCI_INT_DMA_END, SDHCI_INT_STATUS);
 			start_addr &= ~(SDHCI_DEFAULT_BOUNDARY_SIZE - 1);
 			start_addr += SDHCI_DEFAULT_BOUNDARY_SIZE;
diff --git a/include/configs/dragonboard410c.h b/include/configs/dragonboard410c.h
index b97c6c4..890752b 100644
--- a/include/configs/dragonboard410c.h
+++ b/include/configs/dragonboard410c.h
@@ -14,6 +14,9 @@
 
 #define CONFIG_MISC_INIT_R /* To stop autoboot */
 
+/* raw initrd support for snappy */
+#define CONFIG_SUPPORT_RAW_INITRD
+
 /* Physical Memory Map */
 #define CONFIG_NR_DRAM_BANKS		1
 #define PHYS_SDRAM_1			0x80000000
@@ -45,6 +48,9 @@
 #define CONFIG_USB_ETHER_MCS7830
 #define CONFIG_USB_ETHER_SMSC95XX
 
+/* vfat support for snappy */
+#define CONFIG_SUPPORT_VFAT
+
 /* Extra Commands */
 #define CONFIG_CMD_ENV
 /* Enable that for switching of boot partitions */
@@ -66,8 +72,8 @@
 
 #define BOOT_TARGET_DEVICES(func) \
 	func(USB, usb, 0) \
-	func(MMC, mmc, 1) \
 	func(MMC, mmc, 0) \
+	func(MMC, mmc, 1) \
 	func(DHCP, dhcp, na)
 
 #include <config_distro_bootcmd.h>
@@ -86,14 +92,15 @@
 "tftp $loadaddr dragonboard/rescue/gpt_both0.bin && "\
 "mmc write $loadaddr 0 43 && " \
 "mmc rescan && "\
-REFLASH(dragonboard/rescue/NON-HLOS.bin, 1)\
-REFLASH(dragonboard/rescue/sbl1.mbn, 2)\
-REFLASH(dragonboard/rescue/rpm.mbn, 3)\
-REFLASH(dragonboard/rescue/tz.mbn, 4)\
-REFLASH(dragonboard/rescue/hyp.mbn, 5)\
-REFLASH(dragonboard/rescue/sec.dat, 6)\
-REFLASH(dragonboard/rescue/emmc_appsboot.mbn, 7)\
-REFLASH(dragonboard/u-boot.img, 8)\
+REFLASH(dragonboard/rescue/sbl1.mbn, 1)\
+REFLASH(dragonboard/rescue/rpm.mbn, 2)\
+REFLASH(dragonboard/rescue/tz.mbn, 3)\
+REFLASH(dragonboard/rescue/hyp.mbn, 4)\
+REFLASH(dragonboard/rescue/sec.dat, 5)\
+REFLASH(dragonboard/rescue/emmc_appsboot.mbn, 6)\
+REFLASH(dragonboard/u-boot.img, 7)\
+REFLASH(dragonboard/system-boot.img, 8)\
+REFLASH(dragonboard/writable.img, 9)\
 "usb stop &&"\
 "echo Reflash completed"
 
@@ -112,9 +119,15 @@ REFLASH(dragonboard/u-boot.img, 8)\
 	"pxefile_addr_r=0x90100000\0"\
 	BOOTENV
 
-#define CONFIG_ENV_IS_NOWHERE
-#define CONFIG_ENV_SIZE			0x2000
+#undef CONFIG_ENV_IS_NOWHERE
+#define CONFIG_ENV_SIZE		SZ_128K
+#define CONFIG_ENV_IS_IN_FAT
+#define CONFIG_FAT_WRITE
+#define FAT_ENV_INTERFACE      "mmc"
+#define FAT_ENV_DEVICE_AND_PART        "0:8"
+#define FAT_ENV_FILE           "uboot.env"
 #define CONFIG_ENV_VARS_UBOOT_CONFIG
+#define CONFIG_SYS_REDUNDAND_ENVIRONMENT
 
 /* Size of malloc() pool */
 #define CONFIG_SYS_MALLOC_LEN		(CONFIG_ENV_SIZE + SZ_8M)
-- 
2.7.4

